scripts/set_baseline_dependency_versions.ts

```diff
@@ -0,0 +1,52 @@
+import { glob } from 'glob';
+import {
+  readPackageJson,
+  writePackageJson,
+} from './components/package-json/package_json.js';
+import semver from 'semver';
+
+/**
+ * This script sets baseline dependency versions in packages in the workspace for allow listed packages.
+ * Baseline versions are defined as minimum version that matches dependency version declaration.
+ *
+ * This script updates content of 'package.json' files. These changes are needed for testing
+ * and shouldn't be committed to the repo.
+ */
+
+const targetDependencyNames = new Set(['aws-cdk', 'aws-cdk-lib']);
+
+const setBaselineDependencyVersions = async (
+  packagePath: string
+): Promise<void> => {
+  const packageJson = await readPackageJson(packagePath);
+  const dependencyManifests: Array<Record<string, string>> = [];
+  if (packageJson.dependencies) {
+    dependencyManifests.push(packageJson.dependencies);
+  }
+  if (packageJson.devDependencies) {
+    dependencyManifests.push(packageJson.devDependencies);
+  }
+  if (packageJson.peerDependencies) {
+    dependencyManifests.push(packageJson.peerDependencies);
+  }
+
+  for (const dependencyManifest of dependencyManifests) {
+    for (const dependencyName of Object.keys(dependencyManifest)) {
+      if (targetDependencyNames.has(dependencyName)) {
+        const version = dependencyManifest[dependencyName];
+        const baselineVersion = semver.minVersion(version)?.version;
```

Comment:
doesn't really matter, but is this ever something besides the version string with the ^ or ~ stripped off?

Reply:
Docs say that this can handle couple of more cases like this one `semver.minVersion(">=1.0.0"); // '1.0.0'` (per https://docs.npmjs.com/cli/v6/using-npm/semver ).
In practice this is effectively stripping prefixes given how we version deps.